backpack/core/derivatives/shape_check.py

```diff
@@ -39,113 +52,143 @@ def check_shape(mat, like, diff=1):
         )
 
 
-def check_same_V_dim(mat1, mat2):
+def _check_same_V_dim(mat1, mat2):
     V1, V2 = mat1.shape[0], mat2.shape[0]
     if V1 != V2:
         raise RuntimeError("Number of vectors changed. Got {} and {}".format(V1, V2))
 
 
-def check_like(mat, module, name, diff=1, *args, **kwargs):
+def _check_like(mat, module, name, diff=1, *args, **kwargs):
     return check_shape(mat, getattr(module, name), diff=diff)
 
 
-def check_like_with_sum_batch(mat, module, name, sum_batch=True, *args, **kwargs):
+def _check_like_with_sum_batch(mat, module, name, sum_batch=True, *args, **kwargs):
     diff = 1 if sum_batch else 2
     return check_shape(mat, getattr(module, name), diff=diff)
 
 
-def same_dim_as(mat, module, name, *args, **kwargs):
+def _same_dim_as(mat, module, name, *args, **kwargs):
     return len(mat.shape) == len(getattr(module, name).shape)
 
 
 ###############################################################################
 #            Decorators for handling vectors as matrix special case           #
 ###############################################################################
-def mat_prod_accept_vectors(mat_prod, vec_criterion):
+def _mat_prod_accept_vectors(
+    mat_prod: Callable[..., Tensor],
+    vec_criterion: Callable[[Tensor, Module, Any, Any], bool],
+) -> Callable[..., Tensor]:
     """Add support for vectors to matrix products.
 
     vec_criterion(mat, module) returns if mat is a vector.
+
+    Args:
+        mat_prod: .
+        vec_criterion: .
```

Comment:
Function that returns if an input is a single vector that must be formatted into a matrix first before processing